### PR TITLE
feat: remove isAjax

### LINKED ISSUE #186: isAjax & acceptJSON
ref plugin: onerror & security

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

`context.isAjax`, `request.isAjax`

##### Description of change

- remove `request.isAjax`
- remove `ctx.roleFailureHandler`

closes #186

### CODE REVIEW

**Comment by fengmk2** [comment]
rebase master

File: app/extend/request.js [hunk 1]
```diff
@@ -124,33 +124,19 @@ module.exports = {
...
+   * detect if response should be json
+   * 1. url path ends with `.json`
+   * 2. response type is set to json
+   * 3. detect by request accept header
    *
    * @member {Boolean} Request#acceptJSON
-   * @since 2.0.0
+   * @since 1.0.0
    */
   get acceptJSON() {
```
Comment (popomore) [inline app/extend/request.js]: 这个内部要保留原来逻辑么

**Comment by dead-horse** [comment]
等 https://github.com/eggjs/egg/pull/296 这个合并再 rebase

File: app/extend/request.js (hunk 1 above)
Comment (dead-horse) [inline app/extend/request.js]: chair 可以覆盖掉吧

**Pull request merged**